http_check/datadog_checks/http_check/http_check.py

```diff
@@ -305,13 +300,9 @@ def check_cert_expiration(
             sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
             sock.settimeout(float(timeout))
             sock.connect((host, port))
-            context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
-            context.verify_mode = ssl.CERT_REQUIRED
-            context.check_hostname = check_hostname
-            context.load_verify_locations(instance_ca_certs)
 
-            if client_cert:
-                context.load_cert_chain(client_cert, keyfile=client_key)
+            context = self.get_tls_context()
+            context.load_verify_locations(instance_ca_certs)
```

Comment:
I wasn't able to directly apply the `overrides` parameter directly to this.

`TlsContextWrapper` uses either `self.instance or {}` for instance config values, but `http_check` isn't updated to use `self.instance` yet. I tried to refactor to use `self.instance` instead, however since `http_check` uses a specific fixture to get certs in tests: https://github.com/DataDog/integrations-core/blob/master/http_check/tests/conftest.py#L41, I felt that this should be a separate PR later on, but we can still get this PR merged in the meantime.